sympy/solvers/ode/lie_group.py

```diff
@@ -0,0 +1,1088 @@
+r"""
+This module contains the implementation of the lie_group hint for
+dsolve. Ideally any first order differential equation can be solved by lie_group.
+It contains various helper functions which apply different heuristics on the given equation
+and returns the solution.
```

Comment:
This is not a good explanation. Many of the functions below have docstrings with some explanations and also references to papers on which this implementation is based. Those references should be included here ideally with some context like "this implements the algorithm from Y but also with heuristics from Z" or something like that.

Reply:
r"""
This module contains the implementation of the lie_group hint for
dsolve. Ideally any first order differential equation can be solved by lie_group.
It implements various heuristics.

References
==========

- abaco1_simple from E.S Cheb-Terrab, L.G.S Duarte and L.A,C.P da Mota, Computer Algebra
Solving of First Order ODEs Using Symmetry Methods, pp. 8

- abaco1_product fromE.S. Cheb-Terrab, A.D. Roche, Symmetries and First Order ODE Patterns,
pp. 7 - pp. 8

- abaco2_similar from E.S. Cheb-Terrab, A.D. Roche, Symmetries and First Order ODE Patterns,
pp. 10 - pp. 12

- abaco2_unique_unknown from E.S. Cheb-Terrab, A.D. Roche, Symmetries and First Order ODE Patterns,
pp. 10 - pp. 12

- abaco2_unique_general from E.S. Cheb-Terrab, A.D. Roche, Symmetries and First Order
ODE Patterns, pp. 10 - pp. 12

- linear from Lie Groups and Differential Equations pp. 327 - pp. 329

- function_sum from E.S. Cheb-Terrab, A.D. Roche, Symmetries and First Order
ODE Patterns, pp. 7 - pp. 8

- bivariate from Lie Groups and Differential Equations pp. 327 - pp. 329

- chi from E.S Cheb-Terrab, L.G.S Duarte and L.A,C.P da Mota, Computer Algebra
Solving of First Order ODEs Using Symmetry Methods, pp. 8

"""
@oscarbenjamin does this look ok?

Reply:
Lots of those references are repeated so the descriptions can be combined. Also this text doesn't really explain anything about what anything in the module does.

Reply:
Basically I have moved all the internal helper functions only and `single.py` has matching code and general solution for the solver. So should I write one-two line for every function at the top because that also will be redundant as every function has its docstring.

Reply:
It's fine that each function has its own docstring. We just need some text that describes what this module contains from a high level.

Reply:
This module contains the implementation of the internal helper functions for the lie_group hint for
dsolve. These helper functions apply different heuristics on the given equation and return the solution. These functions are used by :py:meth:`sympy.solvers.ode.single.LieGroup`

References
=========

* ``abaco1_simple`` , `function_sum` and `chi`  are referenced from E.S Cheb-Terrab, L.G.S Duarte and L.A,C.P da Mota, Computer Algebra Solving of First Order ODEs Using Symmetry Methods, pp. 8

* `abaco1_product`, `abaco2_similar`, `abaco2_unique_unknown`  and `abaco2_unique_general` are referenced from E.S. Cheb-Terrab, A.D. Roche, Symmetries and First Order ODE Patterns, pp. 7 - pp. 12

* `linear` and `bivariate` from Lie Groups and Differential Equations pp. 327 - pp. 329

does this look better?

Reply:
Yes, that's better. Are there no authors listed for the third reference?

Reply:
No, The author isn't mentioned